content/chainguard/administration/cloudevents/image-copy-gcr/index.md

```diff
@@ -0,0 +1,196 @@
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the file you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you’re satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration. 
+
+```
+. . .
+
+Plan: 8 to add, 0 to change, 0 to destroy.
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+
+. . .
+
+Apply complete! Resources: 8 added, 0 changed, 0 destroyed.
+```
+
+Assuming all the resources were created as expected, you can observe the application in action.
+
+
+## Testing the application
+
+If the `terraform apply` command you ran in the previous section was successful, the Terraform configuration will have set up a Cloud Run service to host the example application. 
+
+As mentioned previously, this application listens for `registry.push` events that occur on your organization's repository; any time a new Image gets added to your organization's Registry the application will mirror it to your GCP project's Artifact Registry and into a repository named with the `name` and `dst_repo` values you set in your `main.tf` file. For example, if these values were `chainguard-dev` and `mirrored`, respectively, (as shown in the previous example) the mirror repository would be found at `<location>-docker.pkg.dev/<project_id>/chainguard-dev-mirrored`.
+
+You can find the results of the application in your GCP Project's dashboard. Navigate to your GCP Project's **Artifact Registry**, then click on the mirror repository you set up with Terraform. There, you will find any Images that have been added to your organization's Registry since you deployed the application. This example shows a repository named `chainguard-dev-mirrored` with two Images (`node` and `python`) mirrored into it.
+
+![Screenshot of a repository in a GCP Artifact Registry named "chainguard-dev-mirrored." This repository shows two Images stored within it. The first, `node`, was created and updated 36 minutes ago, while the second, `python`, was created and last updated 33 minutes ago.](gcp-events-1.png)
+
+Be aware that just because the application is listening for `registry.push` events doesn't mean any will occur automatically. Chainguard Images are generally updated at least once every twenty four hours, so Images may not immediately appear in your mirror repository.
+
+
+## Removing sample resources
+
+If you'd like to remove the resources you created with Terraform, you can run the `terraform destroy` command.
+
+```sh
+terraform destroy
```

Comment:
also thanks for adding this bit here, I was worried to not delete the wrong things when turning it down :D